packages/happy-dom/src/nodes/html-label-element/HTMLLabelElement.ts

```diff
@@ -66,4 +70,23 @@ export default class HTMLLabelElement extends HTMLElement implements IHTMLLabelE
 	public cloneNode(deep = false): IHTMLLabelElement {
 		return <HTMLLabelElement>super.cloneNode(deep);
 	}
+
+	/**
+	 * @override
+	 */
+	public override dispatchEvent(event: Event): boolean {
+		const returnValue = super.dispatchEvent(event);
+
+		if (
+			event.type === 'click' &&
+			(event.eventPhase === EventPhaseEnum.atTarget || event.eventPhase === EventPhaseEnum.bubbling)
+		) {
+			const control = this.control;
+			if (control && event.target !== control) {
+				control.dispatchEvent(new PointerEvent('click', { bubbles: true, cancelable: true }));
```

Comment:
Is this what happens in native browsers as well? It dispatches a click event on the input as well?

Reply:
Yup ok, looks like it is the correct behavior! https://codesandbox.io/s/quizzical-cerf-mp2xtk?file=/src/App.js

https://github.com/capricorn86/happy-dom/assets/168046/3ad2c2fc-98db-4f32-b655-5b35f44239f3

Reply:
Took sometime to figure it out, but it should be correct now :slightly_smiling_face:

Reply:
The label will have two click event dispatched on it if it is a parent to the input, as the event bubbles up to it.

Reply:
Yes sure, it supports that as well.